internal/resource/config/authenticationselector/authentication_selector_resource.go

```diff
@@ -0,0 +1,353 @@
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.ParentRef) {
+		addRequest.ParentRef = &client.ResourceLink{}
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.ParentRef, false)), addRequest.ParentRef)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+
+}
+
+// Metadata returns the resource type name.
+func (r *authenticationSelectorResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_authentication_selector"
+}
+
+func (r *authenticationSelectorResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func (r *authenticationSelectorResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	var plan, state *authenticationSelectorResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	var respDiags diag.Diagnostics
+
+	if plan == nil || state == nil {
+		return
+	}
+
+	plan.Configuration, respDiags = pluginconfiguration.MarkComputedAttrsUnknownOnChange(plan.Configuration, state.Configuration)
+	resp.Diagnostics.Append(respDiags...)
+
+	resp.Plan.Set(ctx, plan)
+}
+
+func readAuthenticationSelectorsResponse(ctx context.Context, r *client.AuthenticationSelector, state *authenticationSelectorResourceModel, configurationFromPlan types.Object) diag.Diagnostics {
+	var diags, objDiags diag.Diagnostics
+
+	state.LastModified = types.StringValue(r.LastModified.Format(time.RFC3339))
+	state.AttributeContract, objDiags = types.ObjectValueFrom(ctx, attributeContractAttrType, r.AttributeContract)
+	diags = append(diags, objDiags...)
+	state.SelectorId = types.StringValue(r.Id)
+	state.Id = types.StringValue(r.Id)
+	state.Name = types.StringValue(r.Name)
+	state.PluginDescriptorRef, objDiags = resourcelink.ToStateNoLocation(&r.PluginDescriptorRef)
+	diags = append(diags, objDiags...)
+	state.ParentRef, objDiags = resourcelink.ToStateNoLocation(r.ParentRef)
+	diags = append(diags, objDiags...)
+	state.Configuration, objDiags = pluginconfiguration.ToState(configurationFromPlan, &r.Configuration)
+	diags = append(diags, objDiags...)
+
+	// make sure all object type building appends diags
+	return diags
+}
+
+func (r *authenticationSelectorResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan authenticationSelectorResourceModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var hasObjectErrMap = make(map[error]bool)
```

Comment:
It looks like you only ever add to this once, so there isn't really a need for the map

Reply:
nice catch